compiler/p/codegen/OMRConstantDataSnippet.cpp

```diff
@@ -155,53 +139,43 @@ int32_t OMR::ConstantDataSnippet::addConstantRequest(void              *v,
       default:
          TR_ASSERT(0, "Only float and address constants are supported. Data type is %s.\n", type.toString());
       }
-
-   return(ret);
    }
 
 
 bool OMR::ConstantDataSnippet::getRequestorsFromNibble(TR::Instruction* nibble, TR::Instruction **q, bool remove)
    {
    ListIterator< PPCConstant<double> >  diterator(&_doubleConstants);
    PPCConstant<double>                 *dcursor=diterator.getFirst();
-   int32_t count = cg()->comp()->target().is64Bit()?4:2;
+   int32_t count;
+
+   if (cg()->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
+      count = 1;
+   else if (cg()->comp()->target().is64Bit())
+      count = 4;
+   else
+      count = 2;
+
+   q[0] = NULL;
+   q[1] = NULL;
+   q[2] = NULL;
+   q[3] = NULL;
 
    while (dcursor != NULL)
       {
       TR_Array<TR::Instruction *> &requestors = dcursor->getRequestors();
-      if (requestors.size() > 0)
+      for (int32_t i = 0; i < requestors.size(); i+=count)
          {
-         for (int32_t i = 0; i < requestors.size(); i+=count)
+         for (int32_t j = 0; j < count; j++)
             {
-            if (count == 2)
+            if (requestors[i + j] == nibble)
                {
-               if (requestors[i] == nibble || requestors[i+1] == nibble)
+               for (int32_t k = 0; k < count; k++)
+                  q[k] = requestors[i + k];
+
+               if (remove)
                   {
-                  q[0] = requestors[i];
-                  q[1] = requestors[i+1];
-                  q[2] = NULL;
-                  q[3] = NULL;
-                  if (remove)
-                     {
-                     requestors.remove(i+1);
+                  for (int32_t k = 0; k < count; k++)
                      requestors.remove(i);
-                     }
-                  return true;
-                  }
-               }
-            else // count == 4
-               {
-               int32_t j;
-               if (requestors[i] == nibble || requestors[i+1] == nibble || requestors[i+2] == nibble || requestors[i+3] == nibble)
-                  {
-                  for (j = 0; j < count; j++)
-                      q[j] = requestors[i+j];
-                  if (remove)
-                     {
-                     for (j = count-1; j >= 0 ; j--)
-                         requestors.remove(i+j);
-                     }
-                  return true;
```

Comment:
Did we lose this` return true `? Here and in similar places below?

Reply:
Fixed in the latest force-push.